specification/integration_context.md

```diff
@@ -0,0 +1,86 @@
+<!-- markdownlint-configure-file 
+{ 
+  "MD013": { 
+    "tables": false 
+  } 
+}
+-->
+
+# Integration Context
+
+**Status**: [Experimental](../README.md#versioning-and-status-of-the-specification)
+
+In order to create a more complete picture of a distributed environment, we
+propose the use of additional headers to pass extra context details between
+Cisco observability components. Specifically, we are concerned with passing
+information between OpenTelemetry-based Splunk distributions and AppDynamics
+agents, in both directions.
+
+# Headers
+
+The following defines several new headers and describes their intended use.
+All headers MUST be treated as optional -- peer services will not always
+generate them.
+
+* `cisco-ctx-acct-id` - Contains the ID of the AppDynamics account.
+* `cisco-ctx-app-id` - Contains the ID of the AppDynamics application.
+* `cisco-ctx-tier-id` - Contains the ID of the AppDynamics tier.
+* `cisco-ctx-bt-id` - Contains the ID of the AppDynamics business transaction (BT).
+* `cisco-ctx-env` - Contains
+  the [deployment.environment.name](https://opentelemetry.io/docs/specs/semconv/attributes-registry/deployment/)
+  resource value from an OpenTelemetry based component.
+* `cisco-ctx-service` - Contains the [service.name](https://opentelemetry.io/docs/specs/semconv/resource/#service)
+  resource value from an OpenTelemetry based component.
+
+HTTP headers are capable of being multivalued. As such, implementations
+SHOULD use the _last_ value when the above headers contain multiple values.
+
+## Splunk OpenTelemetry distributions
+
+This section applies to Splunk distributions of OpenTelemetry instrumentation
+components. In brief, Splunk OTel distributions will need to understand when
+spans are being created as part of an AppDynamics "Business Transaction" (BT),
+and will add span attributes to clarify the AppDynamics invocation context.
+
+### Configuration
+
+Splunk OTel distributions MUST provide a way for users to opt-into the
+consumption and propagation of additional Cisco (bespoke) integration context.
+
+The configuration SHOULD be named `cisco.ctx.enabled` or an idiomatic
+equivalent for each language. If using environment variables, the
+configuration SHOULD be named `CISCO_CTX_ENABLED`. The default value
+MUST be `false` or equivalent language-specific non-truthy value.
+
+## Incoming State
+
+When `cisco.ctx.enabled` is `true`, Splunk implementations MUST
+extract fields from the `cisco-ctx-*` headers (above) and add extra
+attributes to any Spans created as part of the incoming request context.
+Null or missing values MUST be handled gracefully by simply
+omitting the span attributes.
+
+| header              | span attribute             | description                                                             | example                           |
```

Comment:
I think that, you are missing here span type. Part of them seems to be pure numbers, should we treat them as int/strings?

Reply:
Yeah, worth calling out that we should treat them as opaque strings.

Reply:
sure, agreed. added it above.